controllers/rule_controller.go

```diff
@@ -133,16 +137,24 @@ func (r *RuleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	configMap := r.RuleConfigmap
-	if rule.Spec.ConfigMapName != nil {
-		configMap = types.NamespacedName{
-			Name:      *rule.Spec.ConfigMapName,
-			Namespace: req.NamespacedName.Namespace,
+	if !r.IsSideCar {
```

Comment:
We don't have to perform this check every time Reconcile loop runs. The sideCar mode doesn't change. It's better to inject a processing strategy to the RuleReconciler.
It would be an interface with two concrete implementations, and we inject one or the other at the bootstrap time.